lib/Service/RoomService.php

```diff
@@ -392,4 +409,101 @@ public function verifyPassword(Room $room, string $password): array {
 			'url' => '',
 		];
 	}
+
+	public function setTimeToLive(Room $room, Participant $participant, int $ttl): void {
+		$event = new ChangeTtlEvent($room, $ttl);
+		$this->dispatcher->dispatch(Room::EVENT_BEFORE_SET_TIME_TO_LIVE, $event);
+
+		$update = $this->db->getQueryBuilder();
+		$update->update('talk_rooms')
+			->set('time_to_live', $update->createNamedParameter($ttl, IQueryBuilder::PARAM_INT))
+			->where($update->expr()->eq('id', $update->createNamedParameter($room->getId(), IQueryBuilder::PARAM_INT)));
+		$update->executeStatement();
+		$jobList = Server::get(IJobList::class);
+		if ($ttl > 0) {
+			$this->ttlSystemMessage($room, $participant, $ttl, 'ttl_enabled');
+			$jobList->add(ApplyTtl::class, ['room_id' => $room->getId()]);
+		} else {
+			$this->ttlSystemMessage($room, $participant, $ttl, 'ttl_disabled');
+			$jobList->remove(ApplyTtl::class, ['room_id' => $room->getId()]);
+		}
+
+		$this->dispatcher->dispatch(Room::EVENT_AFTER_SET_TIME_TO_LIVE, $event);
+	}
+
+	private function ttlSystemMessage(Room $room, Participant $participant, int $ttl, string $message): void {
+		$this->chatManager->addSystemMessage(
+			$room,
+			$participant->getAttendee()->getActorType(),
+			$participant->getAttendee()->getActorId(),
+			json_encode([
+				'message' => $message,
+				'parameters' => ['ttl' => $ttl]
+			]),
+			$this->timeFactory->getDateTime(),
+			false
+		);
+	}
+
+
+	public function deleteExpiredTtl(int $roomId, int $jobId): array {
+		$room = $this->manager->getRoomById($roomId);
+
+		$max = $this->getMaxDateTtl($room->getTimeToLive());
+		$min = $this->getMinDateTtl($jobId);
+
+		$ids = $this->commentsManager->getMessageIdsByRoomIdInDateInterval($roomId, $min, $max);
+		if (count($ids)) {
+			$this->reportDeletedMessagesIds($room, $ids);
+			$this->deleteMessagesByIds($ids);
+		}
+		return $ids;
+	}
+
+	private function getMaxDateTtl(int $ttl): \DateTime {
+		$max = $this->timeFactory->getDateTime();
+		return $max->sub(new DateInterval('PT' . $ttl . 'S'));
+	}
+
+	private function getMinDateTtl(int $jobId): \DateTime {
+		$query = $this->db->getQueryBuilder();
+		$query->select('last_checked')
+			->from('jobs')
+			->where(
+				$query->expr()->eq('id', $query->createNamedParameter($jobId, IQueryBuilder::PARAM_INT))
+			);
+		$result = $query->executeQuery();
+		$lastCheckedEpoch = $result->fetchOne();
+		$lastChechedDateTime = $this->timeFactory->getDateTime('@' . $lastCheckedEpoch);
+		return $lastChechedDateTime;
+	}
```

Comment:
Illegal access of private table.

Also we shouldn't need it.

As discussed we will change the logic and save the ttl while writing a message and then we can simply get the messages by just checking if their expiration timestamp is != 0 and lower than "NOW"

Reply:
A problem to save the time interval or a date to expire the message is when the room manager change or disable the interval. In this case, if the room manager disable the time to expire and have a message with time to expire in the future, this message will be deleted in the future. If this isn't a problem, we can set by message the time to expire.

Reply:
we will label it like this and it's how other messengers handle it as well. only the TTL while posting is accepted. Whatever applies at that time will take care of this message